blueocean-github-pipeline/src/main/java/io/jenkins/blueocean/blueocean_github_pipeline/GithubEnterpriseScm.java

```diff
@@ -30,14 +35,49 @@ public GithubEnterpriseScm(Reachable parent) {
     }
 
     @Override
+    public @Nonnull String getUri() {
+        String apiUri = getCustomApiUri();
+
+        // NOTE: GithubEnterpriseScm requires that the apiUri be specified
+        if (StringUtils.isEmpty(apiUri)) {
+            throw new ServiceException.BadRequestException(new ErrorMessage(400, "apiUrl is required parameter"));
+        }
+
+        return apiUri;
+    }
+
+    @Override
+    public String getCredentialId() {
+        return createCredentialId(getUri());
+    }
+
+    @Override
     public String getCredentialDomainName() {
-        java.net.URI uri;
-        try {
-            uri = new URI(getUri());
-        } catch (URISyntaxException e) {
-            throw new ServiceException.UnexpectedErrorException(new ErrorMessage(400, "Invalid Github Enterprise URI: "+getUri()));
+        return DOMAIN_NAME;
+    }
+
+    @WebMethod(name="") @GET @TreeResponse
+    public Object getState() {
+        // will enforce that the apiUrl parameter has been sent
+        String apiUri = getUri();
+
+        User user = getAuthenticatedUser();
+        List<StandardUsernamePasswordCredentials> credentials = CredentialsUtils.findCredentials(StandardUsernamePasswordCredentials.class, user, DOMAIN_NAME);
```

Comment:
You can do `CredentialsUtils.findCredential(createCredentialId(apiUrl), StandardUsernamePasswordCredentials.class, new BlueOceanDomainRequirement())`, this will save you from iterating over credentials and finding the right one.